tests/devices/test_default_qutrit.py

```diff
@@ -1391,3 +1396,546 @@ def test_density_matrix_all_wires(self, qutrit_device_2_wires, wires, expected):
         qutrit_device_2_wires.apply(ops)
 
         assert np.allclose(qutrit_device_2_wires.density_matrix(wires), expected)
+
+
+# pylint:disable=no-member
+
+
+# JAX integration tests
+@pytest.fixture(scope="class")
+def jax_import_init(request):
+    request.cls.jax = pytest.importorskip("jax")
+    request.cls.jnp = pytest.importorskip("jax.numpy")
+    request.cls.config = pytest.importorskip("jax.config").config
+
+
+@pytest.mark.jax
+@pytest.mark.usefixtures("jax_import_init")
+@pytest.mark.parametrize('use_jit', [True, False])
+class JaxIntegrationTest:
+    pass
```

Comment:
Can you please give a little more detail on what would be ideal, I'm not sure I follow.

Reply:
Oh I think I understand what you mean.

Basically, let's just explicitly mark `TestQNodeIntegrationJax` with the various markers and not use inheritance.


In source code, extraction of logic and code reuse can make code easier to read and maintain. When reading, you can keep fewer things in your mind at once and think at the same level of abstraction.

But in tests, we want to prioritize debug-ability.  If a test fails, how many definitions do I need to copy and paste into a jupyter notebook? I don't want to go digging through a 1,000+ line file for the definition of more than one or two things.

So my recommendations would be:

1. Directly marking each class with `@pytest.mark.jax`
2. Directly marking each class with `@pytest.mark.parametrize('use_jit', [True, False])`
3. Performing the imports inside the individual tests.


If the test is marked `jax`, then it will only run if jax is installed. So you don't need `importorskip` on interface-specific imports.

Reply:
Oh I understand, thank you!

Reply:
Added